wagtail/admin/views/generic/models.py

```diff
@@ -688,12 +688,18 @@ def form_invalid(self, form):
         return super().form_invalid(form)
 
 
-class CopyView(CreateView):
+class CopyViewMixin:
     def get_object(self, queryset=None):
-        return get_object_or_404(self.model, pk=unquote(str(self.kwargs["pk"])))
+        return get_object_or_404(
+            self.model, pk=unquote(str(self.kwargs[self.pk_url_kwarg]))
+        )
```

Comment:
Extract this into a mixin so the snippets view can just mix this into its `CreateView`.

Would be nice to unify this along with how we do our "detail" views, which currently does this in different ways...

<details>
<summary>Comparison</summary>

`EditView` falls back into the first positional arg if the URL conf does not provide the `pk_url_kwarg`:

https://github.com/wagtail/wagtail/blob/a09bba67cd58f519f3ae5bff32575e7ce9244031/wagtail/admin/views/generic/models.py#L737-L741

`DeleteView` has an optimisation since it loads the object so soon in `setup()`:
https://github.com/wagtail/wagtail/blob/a09bba67cd58f519f3ae5bff32575e7ce9244031/wagtail/admin/views/generic/models.py#L951-L958

`InspectView`:
https://github.com/wagtail/wagtail/blob/a09bba67cd58f519f3ae5bff32575e7ce9244031/wagtail/admin/views/generic/models.py#L1054-L1061

`RevisionCompareView`:
https://github.com/wagtail/wagtail/blob/a09bba67cd58f519f3ae5bff32575e7ce9244031/wagtail/admin/views/generic/models.py#L1183-L1191

`UnpublishView`:
https://github.com/wagtail/wagtail/blob/a09bba67cd58f519f3ae5bff32575e7ce9244031/wagtail/admin/views/generic/models.py#L1286-L1298

`RevisionUnscheduleView`:
https://github.com/wagtail/wagtail/blob/a09bba67cd58f519f3ae5bff32575e7ce9244031/wagtail/admin/views/generic/models.py#L1394-L1404

</details>

#10746 and #11590 would be something to consider as well. Although, I'm wondering, if we're going to refactor `PermissionCheckedMixin` to make use of `ModelPermissionTester` when possible (to allow instance-specific application-level permission checks), how would that affect where we put db call...